### ISSUE TITLE
New fixer conflict between PEAR/FunctionCallSignature and Generic/ScopeIndent

### ISSUE THREAD

**Issue by jrfnl** [squizlabs/PHP_CodeSniffer#1759]
Noticed a new fixer conflict which seems to have been introduced by commit https://github.com/squizlabs/PHP_CodeSniffer/commit/1dc872963a99a17fa9372811f91a7cc1d6d9611b - the `PEAR.Functions.FunctionCallSignature` and `Generic.WhiteSpace.ScopeIndent` sniff end up fighting since that commit.

The issue can be reproduced using this code snippet with the `PEAR` standard:
```php
$deprecated_functions = [
    'the_category_ID'
       => function_call( // 7 spaces, not 8. This is the problem line.
        $a,
        $b
    ),
];
```

**Comment by gsherwood** [comment]
Those sniffs were fighting before this change as well, except in the other direction (larger indents). I think the solution lies in changing the FunctionCallSignature sniff, but not sure exactly how yet.

**Comment by gsherwood** [comment]
I've committed a change to the FunctionCallSignature sniff to enforce an exact tab stop indent on the opening statement (normally the name of the function, but sometimes other tokens or inline html) so that the closing parenthesis indent matches.

The fixer still notices that two sniff want to change the same token (the whitespace indent) so it does an extra loop without fixing anything, but the conflict is gone after that and the fixes can be applied normally. The code example ends up being fixed as:
```php
$deprecated_functions = [
    'the_category_ID'
    => function_call( // 7 spaces, not 8. This is the problem line.
        $a,
        $b
    ),
];
```
i.e., the indent of the double arrow is adjusted down to the tab stop at column 4 and the rest of the function is already indented correctly, so it is left alone.

I changed the way the function indent is calculated to help me apply a fix to it, so it would be great if you would be able to test out the new version of this sniff and see if you think it is still working for you.

**Comment by jrfnl** [comment]
Tested & I can confirm the improved fix works as expected. Thank you!

**Comment by jrfnl** [comment]
Regarding fixer conflicts, it would be great if some of the open PRs regarding these could be tagged for the 3.2.0 release.

Of the series I opened in the last month: 
* There are five which fix PHP errors encountered (fatal errors/undefined index notices etc) which are mostly small isolated PRs: #1707, #1708, #1713 (very small change), #1716, and #1706 (includes additional fixes)
* These two just add `fixed` files, nothing else: #1711, #1723
* These are small fixes (easy to review): #1710, #1712, #1714, #1739
* Very small fix (easy to review), but also adds missing `fixed` file: #1738
* These are the larger/more complex ones: #1717, #1720 (includes adding missing `fixed` file), #1721

**Comment by gsherwood** [comment]
I *was* leaving those for a future release due to the sheer volume of them :) But all those besides the last 3 were really easy to review and merge, so I've done that now, and thanks for all that effort.

I want to try and get 3.2.0 out because it's getting really big, so I've put the last 3 into the 3.3.0 release, along with a few new sniffs from 3.2.0.

Thanks for testing this change as well.

**Comment by jrfnl** [comment]
> I was leaving those for a future release due to the sheer volume of them :) 

@gsherwood I was afraid of that. Glad to see that pointing out the "easy" ones helped to get passed that :smile: Thanks a lot for taking the time & merging these. Cleans up the open PR list.

> But all those besides the last 3 were really easy to review and merge, so I've done that now

I believe you still missed a few of the easy ones, except:
> These are small fixes (easy to review): #1710, #1712, #1714, #1739

I'll run a new fixer conflict check after `3.2.0` has come out. If I remember correctly, there should still be some 50-something files running into fixer conflicts to solve, though the majority of these can probably be fixed in one go (same conflict in multiple files).

**Comment by jrfnl** [comment]
@gsherwood Thanks for merging the additional small ones!
